agent-inject/agent/config.go

```diff
@@ -119,6 +120,7 @@ func (a *Agent) newConfig(init bool) ([]byte, error) {
 		ExitAfterAuth: init,
 		Vault: &VaultConfig{
 			Address:       a.Vault.Address,
+			ProxyAddress:  a.Vault.ProxyAddress,
```

Comment:
I was looking in the [Vault documentation but do not see this value](https://www.vaultproject.io/docs/agent). Below the HTTPS_PROXY environment variable is being said, so this seems redundant anyway. Thoughts?

Reply:
I believe this adds support for configuring the proxy address via the configuration file as well. If we choose to include this, we'd have to update that documentation accordingly.

If you don't feel this particular change is necessary, I can drop this and only allow the proxy address configuration to be set via a command line flag and/or the `AGENT_INJECT_PROXY_ADDR` environment variable.

Reply:
We'd also probably tag this field with `omitempty` as well if we choose to maintain it, as this setting is optional.

Reply:
Did I answer your question? Which action do you suggest? 

I can go ahead with your initial suggestion and remove this bit.

Reply:
`AGENT_INJECT_PROXY_ADDR` sets a global default that will be included with every Vault Agent container. If this is not set, the proxy annotation can be used by the user to explicitly set (or override the global).

What I'm trying to clear up is where this setting ultimately lands. Currently its set in two places: the containers environment variable (HTTPS_PROXY) and the Vault Agent configuration file. It should be one or the other, not both. I wasn't able to find references to it in the Vault Agent config settings documentation, so I think the `HTTPS_PROXY` environment variable is the way to go.

Reply:
You are correct. I was confusing the `AgentConfig` struct as being something like an object for carrying the injector configuration, but it's not.

I removed that property in 7cecb13.